wicker/plugins/wandb.py

```diff
@@ -0,0 +1,95 @@
+import os
+from typing import Any, Dict, Literal
+
+import wandb
+
+from wicker.core.config import get_config
+from wicker.core.definitions import DatasetID
+from wicker.core.storage import S3PathFactory
+
+
+def version_dataset(
+    dataset_name: str,
+    dataset_version: str,
+    metadata: Dict[str, Any],
+    dataset_backend: Literal["s3"] = "s3",
+) -> None:
+    """
+    Version the dataset on Weights and Biases using the config parameters defined in wickerconfig.json.
+
+    Args:
+        dataset_name: The name of the dataset to be versioned
+        dataset_version: The version of the dataset to be versioned
+        metadata: The metadata to be logged as an artifact, enforces dataclass for metadata documentation
+        dataset_backend: The backend where the dataset is stored, currently only supports s3
+    """
+    # needs to first acquire and set wandb creds
+    # WANDB_API_KEY, WANDB_BASE_URL, WANDB_USER_EMAIL
+    _acquire_wandb_credentials()
+
+    # needs to init the wandb run, this is going to be a 'data' run
+    dataset_run = wandb.init(project=f"{dataset_name}_curation", name=dataset_name)
+
+    # grab the uri of the dataset to be versioned
+    dataset_uri = _identify_s3_url_for_dataset_version(dataset_name, dataset_version, dataset_backend)
+
+    # establish the artifact and save the dir/s3_url to the artifact
+    data_artifact = wandb.Artifact(f"{dataset_name}_{dataset_version}", type="dataset")
+    data_artifact.add_reference(dataset_uri, name="dataset")
+
+    # save metadata dict to the artifact
+    data_artifact.metadata["version"] = dataset_version
+    data_artifact.metadata["s3_uri"] = dataset_uri
+    for key, value in metadata.items():
+        data_artifact.metadata[key] = value
+
+    # save the artifact to the run
+    dataset_run.log_artifact(data_artifact)  # type: ignore
+
+
+def _acquire_wandb_credentials() -> None:
```

Comment:
nit: consider renaming to `_set_wandb_credentials` as it sets env vars

Reply:
I like your name better, changed